src/client/dfuse/dfuse_core.c

```diff
@@ -432,6 +468,102 @@ dfuse_pool_open(struct dfuse_projection_info *fs_handle, uuid_t *pool,
 	return rc;
 }
 
+#define ATTR_COUNT 5
+
+char const *const
+cont_attr_names[ATTR_COUNT] = {"dfuse-attr-time",
+			       "dfuse-dentry-time",
+			       "dfuse-dentry-dir-time",
+			       "dfuse-ndentry-time",
+			       "dfuse-data-cache"};
+
+/* Setup caching attributes for a container.
+ *
+ * These are read from pool attributes, or can be overwritten on the command
+ * line, but only for the root dfc in that case, so to use caching with
+ * multiple containers it needs to be set via attributes.
+ *
+ * Returns a  error code on error, or ENODATA if no attributes are
+ * set.
+ */
+static int
+dfuse_cont_get_cache(struct dfuse_cont *dfc)
+{
+	size_t		size;
+	char		*buff;
+	int		rc;
+	int		i;
+	unsigned int	value;
+	bool		have_dentry = false;
+	bool		have_dentry_dir = false;
+	bool		have_attr = false;
+
+	D_ALLOC(buff, 128);
+	if (buff == NULL)
+		return ENOMEM;
+
+	for (i = 0; i < ATTR_COUNT; i++) {
+		size = 128;
+
+		rc = daos_cont_get_attr(dfc->dfs_coh, 1, &cont_attr_names[i],
+					(void * const*)&buff,
+					&size, NULL);
+		if (rc == -DER_NONEXIST) {
+			continue;
+		} else if (rc != -DER_SUCCESS) {
+			DFUSE_TRA_WARNING(dfc, "Failed to load value for '%s' "
+					  DF_RC, cont_attr_names[i], DP_RC(rc));
+			D_GOTO(out, rc = daos_der2errno(rc));
+		}
+		have_attr = true;
+
+		if (i == 4) {
```

Comment:
it's better to have an strcmp here instead of the name to "dfuse-data-cache"

Reply:
Yes, this isn't idea as is.  I'll add a #define index.